JoinColony/colonyCDapp
# [Notifications] feat: toggle notifications muted for a colony

## Description

Added the toggle button in the dropdown menu on the colony homepage.
`DropdownMenu` needs to get refactored since for displaying that menu item, I had to create a **hook** :tada: which returns config data depending on the muted status and loading status.  Kinda overhead.

[Auth proxy PR](https://github.com/JoinColony/colonyCDappAuthProxy/pull/28)

## Testing
Before testing this, ping me or @davecreaser to give you the env secrets in order to test this.
Create a new .env file in the `createUserNotificationsData` with 
```
MAGICBELL_API_KEY=
MAGICBELL_API_SECRET=
```

1. Open up the app as `leela` (or any user), run the following query to assure you don't have any muted colonies yet:
```
query MyQuery {
  getUser(id: "0xb77D57F4959eAfA0339424b83FcFaf9c15407461") {
    id
    profile {
      displayName
    }
    notificationsData {
      magicbellUserId
      mutedColonyIds
      userAddress
    }
  }
}
```
![image](https://github.com/user-attachments/assets/dfe39d76-6c65-472a-badc-30db31f5e90b)
2. Throttle your network to 3G, open the dropdown menu to the right and notice the `Mute notifications` item. Click it, it should change into `Muting notifications` and later into `Unmute notifications`. THe button should stay disabled throughout the entire ordeal.
![image](https://github.com/user-attachments/assets/b9627e75-6fa4-426a-93b8-bbb61d9de8ba)
![image](https://github.com/user-attachments/assets/49c24453-308a-4f6f-b711-5eb708cf28b9)
![image](https://github.com/user-attachments/assets/501f83b7-4304-4ed9-9915-7fc031765c1f)
3. Run the query to be sure it's added to the backend
![image](https://github.com/user-attachments/assets/bdb18c13-824b-4ae0-b583-dc36f39537a5)
4. Do it the other way around now!
![image](https://github.com/user-attachments/assets/689e9901-a60a-496a-ad7f-131d2ea3e6ae)
![image](https://github.com/user-attachments/assets/6b03fe0d-085d-471e-86ef-73500f4313f8)
![image](https://github.com/user-attachments/assets/9c92a794-0e8a-4d65-ac8f-30a599c26ee5)
5. Run the query and be sure the array is now empty
![image](https://github.com/user-attachments/assets/611fc699-067c-4c5b-8e20-3b664016a365)


## Diffs

**New stuff** ✨

* `useMuteItem` hook for displaying the muted item in a smaller hook
* `mutedColonyIds` to the user notification object

**Changes** 🏗

* `mutedColonyIds` to `NotificationContext`

Resolves  #3005
